### PR TITLE
Handle SIP applications that have expired

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Render saved-in-progress applications that have expired differently in the `ApplicationsInProgress` component. Also allow them to be removed.

I took an "optimistic update' approach when a user removes an expired application. That is to say I am removing the box from the UI immediately before we get confirmation that the call to `DELETE /v0/in_progress_forms/{form_id}` has succeeded. The drawbacks to this are pretty minimal; they'll just see the expired form on their dashboard the next time they load the page. But the benefits in terms of UX and development time are nice, IMO. I'll loop up with UX on this. **Update**: [They are cool with this](https://dsva.slack.com/archives/C909ZG2BB/p1614711128127700?thread_ts=1614700220.117800&cid=C909ZG2BB)

## Testing done
Local + updated unit tests for the `ApplicationInProgress` component + updated Cypress tests

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx [hunk 1]
```diff
@@ -3,48 +3,118 @@ import PropTypes from 'prop-types';
...
 import { recordDashboardClick } from '~/applications/personalization/dashboard/helpers';
 
+/**
+ * Returns a copy of the passed-in string with the first letter capitalized
+ *
+ * @param {string} input - word to capitalize
+ * @returns {string} - the input string with the first letter capitalized
+ */
+const capitalizeFirstLetter = input => {
+  const capitalizedFirstLetter = input[0].toUpperCase();
```
Comment (sanlouise) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx:13]: This works, but we could also accomplish this with css using `text-transform: capitalize`

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.unit.spec.jsx [hunk 2]
```diff
@@ -1,38 +1,81 @@
...
+      lastOpenedDate: 'Jan 1, 2019',
+      presentableFormId: 'Form 1234',
+      removeForm: () => {},
+      startNewApplicationUrl: 'application-url',
+    };
   };
-  let view;
-  beforeEach(() => {
-    view = render(<ApplicationInProgress {...props} />);
+  const oneWeekInMs = 7 * 24 * 60 * 60 * 1000;
```
Comment (sanlouise) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.unit.spec.jsx:20]: Ha, I'm wondering if we should move these computed dates into a higher level folder here soon. We're using them a lot in our tests now 😆

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx [hunk 3]
```diff
@@ -3,48 +3,118 @@ import PropTypes from 'prop-types';
...
+          <>
             <div>
-              <p className="vads-u-margin-top--0">
-                Application expires on: {expirationDate}
+              <p className="vads-u-text-transform--uppercase vads-u-margin-y--0">
+                {presentableFormId}
               </p>
-              <p>Last opened on: {lastOpenedDate}</p>
+              <h4 className="vads-u-font-size--h3 vads-u-margin-top--0">
+                {capitalizeFirstLetter(formTitle)}
```
Comment (sanlouise) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx:45]: Does the design still look good when the `formTitle` is long?

**Review by sanlouise — APPROVED** [review]

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx (hunk 3 above)
Comment (erikphansen) [reply src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx:45]: Yeah, they are all long 😆  It just wraps to three or four lines but all the cards look nice next to each other.

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx (hunk 1 above)
Comment (erikphansen) [reply src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx:13]: 🤦 

Oh wait, that will capitalize all the words (ie dumb title casing)

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.unit.spec.jsx (hunk 2 above)
Comment (erikphansen) [reply src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.unit.spec.jsx:20]: Yeah. This should really live in platform, IMO.

**Pull request merged**